libsolidity/parsing/Token.cpp

```diff
@@ -151,20 +153,19 @@ tuple<Token::Value, unsigned int, unsigned int> Token::fromIdentifierOrKeyword(s
 				positionM < positionX &&
 				positionX < _literal.end() &&
 				*positionX == 'x' &&
-				all_of(positionX + 1, _literal.end(), ::isdigit)
+				all_of(++positionX, _literal.end(), ::isdigit)
```

Comment:
correct in the sense that they work. See below the reasoning...However....I can change it back to what it was before with little pain, so I think that can be done.